src/main/java/com/upserve/uppend/lookup/LookupData.java

```diff
@@ -367,22 +366,26 @@ private Long findValueFor(LookupKey key) {
     }
 
     LookupMetadata loadMetadata() {
+        return loadMetadata(null);
+    }
+
+    LookupMetadata loadMetadata(LookupMetadata prev) {
         if (readOnly) {
```

Comment:
Get the two long adders for missCount and hitCount at the top of the method depending on the arguments.
```java
LookupMetadata loadMetadata() {
  return loadMetadata(getMetaHitCount(), getMetaMissCount())
}
LookupMetadata loadMetadata(LookupMetadata prev) {
  return loadMetadata(prev.hitCount, prev.missCount)
}

LookupMetadata loadMetadata(LongAdder hitCount, LongAdder missCount) {
 ...
}
```
I think that might simplify things a bit

Reply:
Done.